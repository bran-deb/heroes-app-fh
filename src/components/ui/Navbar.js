import React, { useContext } from 'react'
import { Link, NavLink } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'

export const Navbar = () => {

    const { user: { name } } = useContext(AuthContext)
    console.log(name)

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid" >

                <Link
                    className="navbar-brand"
                    to="/"
                >
                    Asociaciones
                </Link>

                <div className="collapse navbar-collapse">
                    <div className="navbar-nav">

                        <NavLink
                            className={({ isActive }) => 'nav-item nav-link ' + (isActive ? 'active' : '')}
                            to="/marvel"
                        >
                            Marvel
                        </NavLink>

                        <NavLink
                            className={({ isActive }) => 'nav-item nav-link ' + (isActive ? 'active' : '')}
                            to="/dc"
                        >
                            DC
                        </NavLink>
                        <NavLink
                            className={({ isActive }) => 'nav-item nav-link ' + (isActive ? 'active' : '')}
                            to="/search"
                        >
                            Search
                        </NavLink>
                    </div>
                </div>

                <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                    <ul className="navbar-nav ml-auto">

                        <span className='nav-item nav-link text-info'>
                            {name}
                        </span>
                        <NavLink
                            className={({ isActive }) => "nav-item nav-link" + (isActive ? 'active' : '')}
                            to="/login"
                        >
                            Logout
                        </NavLink>

                    </ul>
                </div>
            </div>
        </nav>
    )
}