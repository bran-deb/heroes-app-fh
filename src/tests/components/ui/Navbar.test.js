import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { AuthContext } from '../../../auth/AuthContext'
import { Navbar } from "../../../components/ui/Navbar"
import { types } from '../../../types/types'

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}))

//usamos el mount por que usamos el context y evaluamos los hijos de un higher order component
describe('Pruebas en <Navbar/>', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Pepe'
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/']}>
                <Navbar />
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('debe de mostrarse correctamente', () => {
        const span = wrapper.find('span')
        expect(wrapper).toMatchSnapshot()
        expect(span.text().trim()).toBe('Pepe')
    })

    test('debe llamar el navigate y el dispatch, con los argumentos', () => {
        wrapper.find('button').prop('onClick')()
        //verifica que el dispatch fue llamado y con que objeto
        expect(contextValue.dispatch).toHaveBeenCalledWith({ 'type': types.logout })
        expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true })
    })
})
