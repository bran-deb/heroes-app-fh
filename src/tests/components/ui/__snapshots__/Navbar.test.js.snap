// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar/> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="collapse navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </div>
</nav>
`;
